ui/core/src/api/SifService/getKeplrProvider.ts

```diff
@@ -46,17 +47,29 @@ type Keplr = {
 // Todo
 type provider = Keplr;
 
+let numChecks = 0;
+
 // Detect mossible keplr provider from browser
-export default (): provider | null => {
+export default async function getKeplrProvider(): Promise<provider | null> {
   const win = window as WindowWithPossibleKeplr;
 
   if (!win) return null;
-
-  if (win.keplr && win.getOfflineSigner) {
-    // assign offline signer (they use __proto__ for some reason), so this is not as pretty as i'd like)
-    Object.getPrototypeOf(win.keplr).getOfflineSigner = win.getOfflineSigner;
-    return win.keplr as Keplr;
+  console.log({
+    "win.keplr": win.keplr,
+    "win.getOfflineSigner": win.getOfflineSigner,
+  });
+
+  if (!win.keplr || !win.getOfflineSigner) {
+    numChecks++;
+    if (numChecks > 20) {
+      return null;
+    }
+    await sleep(100);
+    return getKeplrProvider();
   }
```

Comment:
Probably a better way to do this but this recursively asynchronously checks for keplr until it finds it or fails.